_internals/software-architecture.md

```diff
@@ -5,129 +5,84 @@ nav_order: 1
+from a PostgreSQL server. This allows the PostgreSQL server to speak more than one language, making
+the setup flexible and easy to manage.
 
-The way Babelfish implements Microsoft SQL Server behavior is by using
-[hooks](/docs/internals/postgresql-hooks). The idea is to make database protocols &ldquo;pluggable&rdquo;,
-which means that PostgreSQL can present itself as a different database. In fact, it
-is even possible to speak more than a single protocol at the same time, making
-the setup more flexible and easy to handle.
+A PostgreSQL database that is running Babelfish listens for connections on two (configurable) TCP 
+ports, in two dialects:
+
+- SQL Server dialect (TDS).
+- PostgreSQL dialect.
+ 
+Once the TDS handler has accepted and processed a request coming in on the TDS port,
+the request is passed on to a PostgreSQL backend. The backend parses the request using a 
+custom parser, modified to understand SQL Server dialect.
 
 <img src="/assets/images/babel_architecture.png" title="System architecture" width="800"/>
 
-Once the TDS handler has accepted the request coming in on the default port 1433,
-and once it has processed it, it is passed on via pltds to a standard PostgreSQL backend.
-PostgreSQL will still be exposed to the client on port 5432 (default).
+There are no significant changes to the PostgreSQL optimizer or the PostgreSQL executor.
 
-To sum it up: default:
+The PostgreSQL core that is distributed with Babelfish has been modified to prepare for trigger processing, 
+transaction handling, and other behaviors that are different in SQL Server. This means that the database 
+server is not 100% identical to a community distribution of PostgreSQL.  
 
-- SQL Server dialect, clients connect to port 1433.
-- PostgreSQL dialect, clients connect to port 5432.
+To find out more about unsupported SQL Server behaviors, check
+out our section about [limitations and compatibility issues](/docs/usage/limitations-of-babelfish).
 
-In other words, a Babelfish enabled database will listen for connections on two
-TCP ports instead of one. Administrators have to take that into account. The ports
-used can be configured using PostgreSQL parameters.
+### Databases vs. schemas
 
-How does that work? The TDS listener captures the requests coming
-from the client and sends them though a parser for the SQL dialect of Microsoft
-SQL Server.  The query will then be handled by a regular PostgreSQL
-backend. The optimizer and the executor will basically be the same.
+PostgreSQL supports the following logical structure of objects:
 
-However, some changes had to be made to the PostgreSQL core to prepare for
-trigger processing, transaction handling and other aspects that are different in
-Microsoft SQL Server.  This means that the database server is not 100% identical
-to what one would expect from PostgreSQL.  Some behavior had to be changed to
-behave like Microsoft SQL Server.  Keep that in mind!
+- Instance: a running PostgreSQL server
```

Comment:
Maybe worth mentioning that in Postgres community documentation they use the term 'cluster'.

Reply:
Now reads:
PostgreSQL supports the following logical structure of objects:

- Instance: a running PostgreSQL server (the PostgreSQL community documentation calls this a cluster)
- Database: what a user connects to
- Schema: a logical grouping of tables and other objects inside a database
- Table: where the data is stored